packages/xrpc-server/src/util.ts

```diff
@@ -182,3 +187,33 @@ export function processBodyAsBytes(req: express.Request): Promise<Uint8Array> {
     req.on('end', () => resolve(new Uint8Array(Buffer.concat(chunks))))
   })
 }
+
+function decodeBodyStream(
+  req: express.Request,
+  maxSize: number | undefined,
+): Readable {
+  const contentEncoding = req.headers['content-encoding']
+  const contentLength = req.headers['content-length']
+
+  if (
+    maxSize !== undefined &&
+    contentLength &&
+    parseInt(contentLength, 10) > maxSize
```

Comment:
doesn't need to hold up this PR, but a bad client could send up fake content-lengths right? is there something we can stick on the stream to bail out & throw an error if it goes over the purported length?

Reply:
Yep yep, you're dead on.  There is a way we could turn that into an error.  There's a similarish case here where a client could send up bad data by saying it's gzipped, but it's actually not— that would cause an error to be emitted from the gzip decoder stream that should turn into a 400.  The issue I am a bit stuck on is that there's not a great way for these errors to surface properly right now— I believe it would become an unhandled exception.  So I can get that error going, but I think there's a bigger question here about how to catch errors off the stream and turn them into appropriate responses.

Reply:
I will at minimum peel this off into an issue prior to merging 👍

Reply:
yeah i think we'd want to emit the error on the stream that we pass to the event handler 🤔

Reply:
This didn't end-up being too bad 👍 Should be handled now!